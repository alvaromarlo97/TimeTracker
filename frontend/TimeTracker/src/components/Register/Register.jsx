/* eslint-disable global-require */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import {
  View,
  TextInput,
  TouchableOpacity,
  Text,
  Image,
} from 'react-native';
import { useDispatch } from 'react-redux';
import styles from './register.style';
import { RegisterUser } from '../../redux/actions/userCreators';
// import store from '../../redux/store';

export default function Register({ navigation }) {
  const dispatch = useDispatch();
  const [nameRInputValue, setNameRInputValue] = useState('');
  const [emailRInputValue, setEmailRInputValue] = useState('');
  const [passwordRInputValue, setPasswordRInputValue] = useState('');

  function handleRegistation() {
    if (nameRInputValue && passwordRInputValue && emailRInputValue) {
      dispatch(RegisterUser({
        username: nameRInputValue,
        email: emailRInputValue,
        password: passwordRInputValue,
      }));
      navigation.push('Login');
    }
  }

  return (
    <View style={styles.container}>
      <Image source={require('../../public/image/timeOrange.png')} style={styles.image} />
      <TextInput
        placeholder="username"
        autoCorrect={false}
        autoCapitalize="words"
        style={styles.email}
        onChangeText={(text) => setNameRInputValue(text)}
        value={nameRInputValue}
      />
      <TextInput
        placeholder="email"
        autoCorrect={false}
        autoCapitalize="words"
        style={styles.email1}
        onChangeText={(text) => setEmailRInputValue(text)}
        value={emailRInputValue}
      />
      <TextInput
        placeholder="password"
        secureTextEntry
        style={styles.email1}
        onChangeText={(text) => setPasswordRInputValue(text)}
        value={passwordRInputValue}
      />
      <TouchableOpacity style={styles.button} onPress={handleRegistation}>
        <Text style={styles.login}>Register</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.button1} onPress={() => navigation.push('Login')}>
        <Text style={styles.login1}>Login</Text>
      </TouchableOpacity>
    </View>
  );
}
