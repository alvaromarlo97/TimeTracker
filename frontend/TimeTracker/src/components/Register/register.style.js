import { StyleSheet } from 'react-native';

export default StyleSheet.create({

  container: {
    flex: 1,
    backgroundColor: '#262155',
    borderRadius: 25,
    display: 'flex',
    alignItems: 'center',
  },
  image: {
    marginTop: 100,
    width: 200,
    height: 200,
  },
  email: {
    fontFamily: 'Avenir-Medium',
    fontSize: 25,
    color: 'white',
    height: 69,
    width: 263,
    backgroundColor: '#61DFE7',
    textAlign: 'center',
    borderRadius: 25,
    marginTop: 100,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,

  },
  email1: {
    fontFamily: 'Avenir-Medium',
    fontSize: 25,
    color: 'white',
    height: 69,
    width: 263,
    backgroundColor: '#61DFE7',
    textAlign: 'center',
    borderRadius: 25,
    marginTop: 15,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,

  },
  button: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: 175,
    height: 69,
    backgroundColor: '#FD9F32',
    borderRadius: 25,
    marginTop: 25,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,

  },
  login: {
    fontFamily: 'Avenir-Medium',
    color: 'white',
    textAlign: 'center',
    alignItems: 'center',
    fontSize: 30,
    marginTop: 8,
  },
  login1: {
    fontFamily: 'Avenir-Medium',
    color: 'white',
    textAlign: 'center',
    alignItems: 'center',
    fontSize: 30,
    marginTop: 8,
  },

});
