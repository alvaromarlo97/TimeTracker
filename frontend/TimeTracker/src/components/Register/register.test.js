import React from 'react';
import { render } from '../../../utils/testUtils/test.utils';
import Register from './Register';

const navigation = {
  navigate: jest.fn(),
};

describe('Given a Character Component', () => {
  describe('When it is rendered', () => {
    let screen;
    beforeEach(() => {
      screen = render(<Register navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});
