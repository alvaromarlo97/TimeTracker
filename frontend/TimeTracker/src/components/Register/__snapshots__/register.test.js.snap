// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#262155",
      "borderRadius": 25,
      "display": "flex",
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/public/image/timeOrange.png",
      }
    }
    style={
      Object {
        "height": 200,
        "marginTop": 100,
        "width": 200,
      }
    }
  />
  <TextInput
    autoCapitalize="words"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="username"
    style={
      Object {
        "backgroundColor": "#61DFE7",
        "borderRadius": 25,
        "color": "white",
        "fontFamily": "Avenir-Medium",
        "fontSize": 25,
        "height": 69,
        "marginTop": 100,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.44,
        "shadowRadius": 5,
        "textAlign": "center",
        "width": 263,
      }
    }
    value=""
  />
  <TextInput
    autoCapitalize="words"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="email"
    style={
      Object {
        "backgroundColor": "#61DFE7",
        "borderRadius": 25,
        "color": "white",
        "fontFamily": "Avenir-Medium",
        "fontSize": 25,
        "height": 69,
        "marginTop": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.44,
        "shadowRadius": 5,
        "textAlign": "center",
        "width": 263,
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="password"
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#61DFE7",
        "borderRadius": 25,
        "color": "white",
        "fontFamily": "Avenir-Medium",
        "fontSize": 25,
        "height": 69,
        "marginTop": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.44,
        "shadowRadius": 5,
        "textAlign": "center",
        "width": 263,
      }
    }
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FD9F32",
        "borderRadius": 25,
        "display": "flex",
        "height": 69,
        "justifyContent": "center",
        "marginTop": 25,
        "opacity": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.44,
        "shadowRadius": 5,
        "width": 175,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "white",
          "fontFamily": "Avenir-Medium",
          "fontSize": 30,
          "marginTop": 8,
          "textAlign": "center",
        }
      }
    >
      Register
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "white",
          "fontFamily": "Avenir-Medium",
          "fontSize": 30,
          "marginTop": 8,
          "textAlign": "center",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
